src/DataWrangling/JRA55.jl

```diff
@@ -140,83 +298,319 @@ Keyword arguments
+                                              latitude = φr,
+                                              topology = (TX, Bounded, Flat))
+
+    boundary_conditions = FieldBoundaryConditions(JRA55_native_grid, (Center, Center, Nothing))
+    times = jra55_times(native_times)
+    
+    if backend isa JRA55NetCDFBackend
+        fts = FieldTimeSeries{Center, Center, Nothing}(JRA55_native_grid, times;
+                                                       backend,
+                                                       time_indexing,
+                                                       boundary_conditions,
+                                                       path = filename,
+                                                       name = shortname)
+
+        # Fill the data in a GPU-friendly manner
+        copyto!(interior(fts, :, :, 1, :), data)
+        fill_halo_regions!(fts)
 
-    # Fill the data
-    interior(fts, :, :, 1, :) .= data[:, :, :]
+        return fts
+    else
+        # Make times into an array for later preprocessing
+        if !totally_in_memory
+            times = collect(times)
+        end
 
-    # Fill halo regions so we can interpolate to finer grids
-    Nt = length(times)
-    chunk_size = 100
-    if Nt <= chunk_size # one chunk will do
-        fill_halo_regions!(fts)
-    else # need multiple chunks
-        start = 1
-        while start < Nt
-            stop = min(Nt, start + chunk_size - 1)
-            fts_chunk = Tuple(fts[n] for n = start:stop)
-            fill_halo_regions!(fts_chunk)
-            start += chunk_size
+        native_fts = FieldTimeSeries{Center, Center, Nothing}(JRA55_native_grid, times;
+                                                              time_indexing,
+                                                              boundary_conditions)
+
+        # Fill the data in a GPU-friendly manner
+        copyto!(interior(native_fts, :, :, 1, :), data)
+        fill_halo_regions!(native_fts)
+
+        @show on_native_grid
+        @show totally_in_memory
+
+        if on_native_grid && totally_in_memory
+            return native_fts
+
+        elseif totally_in_memory # but not on the native grid!
+            boundary_conditions = FieldBoundaryConditions(grid, (LX, LY, Nothing))
+            fts = FieldTimeSeries{LX, LY, Nothing}(grid, times; time_indexing, boundary_conditions)
+            interpolate!(fts, native_fts)
+            return fts
         end
     end
 
-    return fts
+    @show totally_in_memory
+    @info "Pre-processing JRA55 $variable_name data into a JLD2 file..."
+
+    on_disk_fts = FieldTimeSeries{LX, LY, Nothing}(fts.grid;
+                                                   boundary_conditions,
+                                                   backend = OnDisk(),
+                                                   path = jld2_filename,
+                                                   name = fts_name)
+
```

Comment:
I guess something like this would work

```suggestion
    preprocessing_grid = on_native_grid ? JRA55_native_grid : grid

    on_disk_fts = FieldTimeSeries{LX, LY, Nothing}(preprocessing_grid;
                                                                                     boundary_conditions,
                                                                                     backend = OnDisk(),
                                                                                     path = jld2_filename,
                                                                                     name = fts_name)

```